sds_data_manager/lambda_images/instruments/batch_starter.py

```diff
@@ -0,0 +1,425 @@
+    process_range : int
+        Numbers of days backwards to process
+
+    Returns
+    -------
+    level : str
+        Instrument level
+    version : int
+        Version
+    process_dates : list
+        Dates to process
+    """
+
+    query = f"""SELECT * FROM sdc.{instrument.lower()}
+                WHERE filename = %s
+                LIMIT 1;"""
+    params = (filename,)
+
+    cur.execute(query, params)
+    column_names = [desc[0] for desc in cur.description]
+    records = cur.fetchall()
+
+    if not records:
+        raise ValueError(f"No records found for filename: {filename}")
+
+    record_dict = dict(zip(column_names, records[0]))
+
+    level = record_dict["level"]
+    version = record_dict["version"]
+
+    dt = record_dict["date"]
+    dt_start = dt - timedelta(days=process_range)
+
+    # Generate all the dates between dt_start and dt, inclusive
+    current_date = dt_start
+    process_dates = []
+
+    while current_date <= dt:
+        process_dates.append(current_date.strftime("%Y-%m-%d"))
+        current_date += timedelta(days=1)
+
+    return level, version, process_dates
+
+
+def query_dependents(cur, version, process_dates, instrument_dependents):
+    """
+    Queries the database for dependent instruments and retrieves their records.
+
+    Parameters
+    ----------
+    cur : psycopg2.extensions.cursor
+        A psycopg2 database cursor object to execute database operations.
+    version : int
+        Version of the instrument to be queried.
+    process_dates : list
+        A list containing start and end date to filter records on their ingestion date.
+    instrument_dependents : list of dict
+        A list containing dictionaries of dependent instruments and their levels.
+        Each dictionary should have keys 'instrument' and 'level'.
+
+    Returns
+    -------
+    all_records : list of dict
+        A list of dictionaries where each dictionary corresponds to a record
+        from the database that matches the given criteria.
+
+    """
+    all_records = []
+
+    # Loop through instrument dependents and query them
+    for instrument in instrument_dependents:
+        query = f"""SELECT * FROM sdc.{instrument['instrument'].lower()}
+                    WHERE version = %s
+                    AND level = %s
+                    AND ingested BETWEEN %s::DATE AND (
+                    %s::DATE + INTERVAL '1 DAY');"""
+        params = (
+            version,
+            instrument["level"].lower(),
+            min(process_dates),
+            max(process_dates),
+        )
+
+        cur.execute(query, params)
+        column_names = [desc[0] for desc in cur.description]
+        records = cur.fetchall()
+
+        # Map the column names to the records
+        records_dicts = [dict(zip(column_names, record)) for record in records]
+
+        all_records.extend(records_dicts)
+
+    return all_records
+
+
+def remove_ingested(records, dependents_level, process_dates):
```

Comment:
It's looking to see if the data product that we are planning on processing has already been ingested. This is just to prevent duplicates, but I am thinking that we might just remove this part of the code. We can get together are discuss next week.